pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -18,140 +18,42 @@ package vsphere_volume
 
 import (
 	"errors"
-	"io/ioutil"
+	"fmt"
 	"os"
-	"path"
 	"strings"
 	"time"
 
 	"github.com/golang/glog"
-	"k8s.io/kubernetes/pkg/util/keymutex"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/pkg/util/mount"
 	"k8s.io/kubernetes/pkg/volume"
 )
 
 const (
-	maxRetries = 10
+	maxRetries         = 10
+	checkSleepDuration = time.Second
+	diskByIDPath       = "/dev/disk/by-id/"
+	diskSCSIPrefix     = "wwn-0x"
 )
 
 var ErrProbeVolume = errors.New("Error scanning attached volumes")
 
-// Singleton key mutex for keeping attach/detach operations for the same PD atomic
-var attachDetachMutex = keymutex.NewKeyMutex()
-
 type VsphereDiskUtil struct{}
 
-// Attaches a disk to the current kubelet.
-// Mounts the disk to it's global path.
-func (util *VsphereDiskUtil) AttachDisk(vm *vsphereVolumeMounter, globalPDPath string) error {
-	options := []string{}
-
-	// Block execution until any pending attach/detach operations for this PD have completed
-	attachDetachMutex.LockKey(vm.volPath)
-	defer attachDetachMutex.UnlockKey(vm.volPath)
-
-	cloud, err := vm.plugin.getCloudProvider()
-	if err != nil {
-		return err
-	}
-
-	diskID, diskUUID, attachError := cloud.AttachDisk(vm.volPath, "")
-	if attachError != nil {
-		return attachError
-	} else if diskUUID == "" {
-		return errors.New("Disk UUID has no value")
-	}
-
-	// diskID for detach Disk
-	vm.diskID = diskID
-
-	var devicePath string
-	numTries := 0
-	for {
-		devicePath = verifyDevicePath(diskUUID)
-
-		_, err := os.Stat(devicePath)
-		if err == nil {
-			break
-		}
-		if err != nil && !os.IsNotExist(err) {
-			return err
-		}
-		numTries++
-		if numTries == maxRetries {
-			return errors.New("Could not attach disk: Timeout after 60s")
-		}
-		time.Sleep(time.Second * 60)
+func verifyDevicePath(path string) (string, error) {
+	if pathExists, err := pathExists(path); err != nil {
```

Comment:
PathExists(...) method in pkg/volume/util/util.go was added recently so had no idea. I have now used this method from pkg/volume/util/util.go and removed it from pkg/volume/vsphere_volume/vsphere_volume_util.go